src/mail.sh

```diff
@@ -263,23 +285,13 @@ function mail_verify()
     fi
   done
 
-  for config in "${optional_config_options[@]}"; do
-    if [[ ! "${!set_confs[*]} " =~ (local|global)_"$config"[[:space:]] ]]; then
-      missing_opt_conf["$count_opt"]="$config"
-      count_opt=$((count_opt + 1))
-    fi
-  done
+  mapfile -t missing_opt_conf < <(missing_options 'optional_config_options')
 
-  for config in "${essential_config_options[@]}"; do
-    if [[ ! "${!set_confs[*]} " =~ (local|global)_"$config"[[:space:]] ]]; then
-      missing_conf["$count"]="$config"
-      count=$((count + 1))
-    fi
-  done
+  mapfile -t missing_conf < <(missing_options 'essential_config_options')
 
-  if [[ "$count" -gt 0 ]]; then
+  if [[ "${missing_conf[0]}" -gt 0 ]]; then
```

Comment:
fwiu, `missing_options` will return the index and fill out the target array. In this sense, I suppose that `${missing_conf[0]}` should have a value or maybe be empty. I did not get why you are comparing this value to 0? Is it correct?

Reply:
Currently `${missing_conf[0]}` stores the number of missing configs, but I suppose I could change it to check for the number of arguments or if it's empty.